### PR TITLE
Bump golangci-lint to v1.47.3

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

This PR bumps golangci-lint to v1.47.3, re-enabling the bodyclose linter.
This PR also:
- disables `structcheck` because it doesn't support go1.18, and because it's deprecated
- disables `package-comments` revive rule, because it triggers more than > 300 times, maybe this can be fixed later. It now triggers because the confidence of this specific revive check was raised and it's now greater than the default confidence threshold for reporting (see https://github.com/mgechev/revive/pull/694)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by djmitche** [comment]
> disables package-comments revive rule

I've been seeing this on new packages or packages that I modify substantially, I think?  I didn't realize there was a variable threshold, but that does explain it.

**Review by djmitche — APPROVED** [review]

**Comment by mx-psi** [comment]
> disables package-comments revive rule, because it triggers more than > 300 times, maybe this can be fixed later. It now triggers because the confidence of this specific revive check was raised and it's now greater than the default confidence threshold for reporting 

Could you create a ticket for this?

File: internal/tools/go.mod [hunk 1]
```diff
@@ -5,27 +5,30 @@ go 1.18
...
+	github.com/mgechev/revive v1.2.3
 	github.com/stormcat24/protodep v0.0.0-20210106175019-c95189bc461b
 	github.com/vektra/mockery/v2 v2.12.3
 	golang.org/x/mobile v0.0.0-20201217150744-e6ae53a27f4f
 	golang.org/x/perf v0.0.0-20210220033136-40a54f11e909
 	gotest.tools/gotestsum v1.8.2
 )
 
+// we want to force this version, since 1.0.6 is broken on go1.18
+replace github.com/sylvia7788/contextcheck => github.com/sylvia7788/contextcheck v1.0.4
```
Comment (mx-psi) [inline internal/tools/go.mod:19]: Is there an issue or link that explains how this is broken and we can link here?

Comment (paulcacheux) [reply internal/tools/go.mod:19]: The issue I'm facing is that go tries very hard to make me install v1.0.6, even though golangci declares 1.0.4 in its go.mod. And it doesn't compile on go1.18. I did not find an issue with the same problem. Could you try to reproduce (basically remove the replace, `inv -e tidy-all; inv -e install-tools`. Thanks !

**Comment by paulcacheux** [comment]
I will create a ticket for `package-comments` once/if this is merged

File: internal/tools/go.mod (hunk 1 above)
Comment (mx-psi) [reply internal/tools/go.mod:19]: I reproduced it locally. AIUI, this is not related to Go 1.18; the issue is that contextcheck made a breaking change in v1.0.6 in their public API (specifically, in the signature of `contextcheck.NewAnalyzer`, which now needs to be passed a config struct), and golangci-lint v1.47.3 still uses the old signature. This was fixed on golangci-lint's side on v1.49.0 (https://github.com/golangci/golangci-lint/commit/cc2d97f31dea62c47c55d5b7a93637e242df7ad1).

We need to either update the comment saying that this needs to be removed on v1.49.0 or just bump to v1.49.0.

Comment (paulcacheux) [reply internal/tools/go.mod:19]: Oh great, let me try bumping to 1.49.0

Comment (paulcacheux) [reply internal/tools/go.mod:19]: I'm getting a `note: module requires Go 1.19` when trying 1.49.0. I will add a comment

Comment (mx-psi) [reply internal/tools/go.mod:20]: 😞 that's a pity, well, let's just add a note then

Comment (paulcacheux) [reply internal/tools/go.mod:20]: done in the last commit

**Review by mx-psi — APPROVED** [review]
LGTM, you will need 041bd2374886603dbdc6bd450cb2c20e626f7e74 for (CircleCI) tests to pass

**Pull request merged**